### DISCUSSION TITLE
Layout with params?

### DISCUSSION THREAD

**Discussion by bruceharrison1984** [TanStack/router#1718] [Q&A]
Is there anyway to create a layout route that also includes params? This would be for hierarchical data, where a parent object could be loaded for all child routes. The parent route would not be directly routable, and the parent is only accessible in the context of a child.

- `layout` / `accessibleRoute`
- `/_parentObject/$parentId` / `childObject/$childId`
- `/_parentObject/$parentId` / `settings`

**Comment by Ahmad-Jamal** [comment]
@bruceharrison1984 did you find any solution ?

**Comment by schiller-manuel** [comment]
Is this what you are looking for?

https://stackblitz.com/edit/tanstack-router-vmrcmc?file=src%2Froutes%2F%24id%2Ffoo.tsx,src%2Froutes%2F%24id%2Fbar.tsx&preset=node
